GeneticsCore/src/org/labkey/GeneticsCore/GeneticsCoreController.java

```diff
@@ -18,4 +32,73 @@ public GeneticsCoreController()
     {
         setActionResolver(_actionResolver);
     }
+
+    @RequiresPermission(AdminPermission.class)
+    public class AggregateMhcAction extends ConfirmAction<AggregateMhcForm>
+    {
+        @Override
+        public ModelAndView getConfirmView(AggregateMhcForm o, BindException errors) throws Exception
+        {
+            setTitle("Sync MHC Data from PRIMe");
+
+            StringBuilder sb = new StringBuilder();
+            sb.append("This will aggregate MHC typing data into the mhc_data table, creating a single dataset with one row per animal/marker. Do you want to continue?");
+            sb.append("<br><br><label>Check this box to reset the last run time: </label> <input type='checkbox' name = 'resetLastRun' />");
+
+            return new HtmlView(HtmlString.unsafe(sb.toString()));
+        }
+
+        @Override
+        public boolean handlePost(AggregateMhcForm o, BindException errors) throws Exception
+        {
+            try
+            {
+                if (o.isResetLastRun())
+                {
+                    MhcAggregationPipelineJob.saveLastRun(getContainer(), null);
+                }
+
+                PipeRoot pipelineRoot = PipelineService.get().findPipelineRoot(getContainer());
+                MhcAggregationPipelineJob job = new MhcAggregationPipelineJob(getContainer(), getUser(), getViewContext().getActionURL(), pipelineRoot);
+                PipelineService.get().queueJob(job);
+            }
+            catch (Exception e)
+            {
+                _log.error(e);
```

Comment:
This will log the toString() of the exception, not the full stack trace. Add a message argument if you want the full stack.

Reply:
> @bbimber before I do that, PRIMe is on 20.11 right now. I don't know your timetable for wanting to deploy this. Is 21.3 the desired branch?

I see - I thought it was newer. That's a tough call. So I only need the one ETL XML file deployed to PRIMe. Most of this, including the pipeline job, runs on prime-seq, which is 21.3. Perhaps I could split this into two PRs that separates the ETL file?

Reply:
On logging: for some reason I thought what I was doing worked. One question, is this safe? I think e.getMessage() is nullable, right? Do I need a null check or would it be better practice to just add my own message?

_log.error(e.getMessage(), e);

vs.

_log.error("Error starting MHC pipeline", e);

Reply:
I prefer the second approach, where you give a message so that the log has a little more context, but it should be safe to pass null without creating an NPE